source/simulator/assembly.cc

```diff
@@ -903,6 +922,36 @@ namespace aspect
                                                         advection_field);
       }
 
+    // if set to true, the maximum of the artificial viscosity in the cell as well
+    // as the neighbors of the cell is computed and used instead
+    if (use_smoothing == true)
+      {
+        Vector<T> viscosity_per_cell_temp;
+        viscosity_per_cell_temp.reinit(triangulation.n_active_cells());
+
+        viscosity_per_cell_temp = viscosity_per_cell;
+        typename DoFHandler<dim>::active_cell_iterator
+        cell,
+        end_cell = dof_handler.end();
+        for (cell = dof_handler.begin_active(); cell!=end_cell; ++cell)
+          {
+            if (cell->is_locally_owned())
+              for (unsigned int face_no=0; face_no<GeometryInfo<dim>::faces_per_cell; ++face_no)
+                {
+                  if (cell->at_boundary(face_no) == false)
+                    if (cell->neighbor(face_no)->active())
+                      viscosity_per_cell[cell->user_index()] = std::max(viscosity_per_cell[cell->user_index()],
```

Comment:
Replace `cell->user_index()` here and 5 lines below by `cell->active_cell_index()`.